sematic/db/models/run.py

```diff
@@ -126,3 +135,20 @@ def strip_description(self, key, value) -> str:
             value = re.sub(r"\n\s{4}", "\n", value.strip())
 
         return value
+
+    @property
+    def resource_requirements(self) -> Optional[ResourceRequirements]:
+        if self.resource_requirements_json is None:
+            return None
+
+        json_encodable = json.loads(self.resource_requirements_json)
+        return value_from_json_encodable(json_encodable, ResourceRequirements)
+
+    @resource_requirements.setter
+    def resource_requirements(self, value: Optional[ResourceRequirements]) -> None:
+        if value is None:
+            self.resource_requirements_json = None
+            return
+        self.resource_requirements_json = json.dumps(
+            value_to_json_encodable(value, ResourceRequirements)
+        )
```

Comment:
Why not use `@validates` to do this? See `convert_tags_to_json`.

Then you could simply pass it to the constructor in `factories.py`.

Reply:
So there's a problem with the pattern that's already in use for fields that get converted from some "python-y" representation to json-string-y rep for these DB things. The issue is that it makes the objects kind of a pain to *read* from in python-space. When I have a `Run` in python, I want to be able to call `run.resource_requirements` and get an instance of the Dataclass. If we just use a `@validates` and name the database field `resource_requirements` without a property like this, then we can *set* `run.resource_requirements` with a dataclass rep, but when we read it back we'll just get a string.

Here's my proposed alternative:
- for the actual columns of the DB, we name them `x_json`. This contains a json-string-y thing
- for the stuff we work with in python, we wrap `self.x_json` with a property that can be used as `self.x = x0` or `self.x.foo`
- the DB column remains the source of truth

Reply:
So there's a problem with the pattern that's already in use for fields that get converted from some "python-y" representation to json-string-y rep for these DB things. The issue is that it makes the objects kind of a pain to *read* from in python-space. When I have a `Run` in python, I want to be able to call `run.resource_requirements` and get an instance of the Dataclass. If we just use a `@validates` and name the database field `resource_requirements` without a property like this, then we can *set* `run.resource_requirements` with a dataclass rep, but when we read it back we'll just get a string.

Here's my proposed alternative:
- for the actual columns of the DB, we name them `x_json`. This contains a json-string-y thing
- for the stuff we work with in python, we wrap `self.x_json` with a property that can be used as `self.x = x0` or `self.x.foo`
- the DB column remains the source of truth